### PR TITLE
Application form

### PR DESCRIPTION
**Pull request by kareemarab**

### CODE REVIEW

**Comment by kareemarab** [comment]
This PR still needs work.
Completed: most of basic form design per figma desgn.
Remaining: 
- layout-ing
- drag & drop (browse only might suffice, check w/ core-design)
- link adding
- wiring up to upload api + data mapping

**Review by tomknig — COMMENTED** [review]
Thank you for working on the application form. I see that you acknowledge the following tasks remain open:
- file upload component // drag & drop functionality for file uploader
- "add link" component
- wiring up to upload api + data mapping

However, I think this implementation deviates quite much from the design. I am not sure how close we want to get to it, but I see many differences, e.g. in terms of:
- background
- color of the pane
- border radiuses
- shadows
- font families, sizes, and weights
- input field appearance
- spacings

Design:
image

Implementation:
image

@kurdybacha and @KevalinKetcham, I'd be curious to hear your oppinions.

**Comment by kareemarab** [comment]
> Thank you for working on the application form. I see that you acknowledge the following tasks remain open:
> 
> * file upload component // drag & drop functionality for file uploader
> * "add link" component
> * wiring up to upload api + data mapping
> 
> However, I think this implementation deviates quite much from the design. I am not sure how close we want to get to it, but I see many differences, e.g. in terms of:
> 
> * background
> * color of the pane
> * border radiuses
> * shadows
> * font families, sizes, and weights
> * input field appearance
> * spacings
> 
> Design: image
> 
> Implementation: image
> 
> @kurdybacha and @KevalinKetcham, I'd be curious to hear your oppinions.

Hey @tomknig, thanks for pointing those things out! I just want to be clear about which version I used. I used the "Current design" in figma as a reference. 

Screen Shot 2021-12-15 at 2 43 16 PM

I agree some things are off, but my aim was to get the ball rolling with the other components w/ this pr as a foundation. If I'm incorrect about the figma reference, then I can go back and make all the changes, if I'm not then I think the list above gets smaller, no?

**Comment by tomknig** [comment]
Interesting. But I agree. The design you are looking at doesn't diverge too far from the implementation, that's fine imo. Are we looking at different figmas maybe? I am in https://www.figma.com/file/CPfZkeob1dtxStjSuUUXDs/HyperscalePaid?node-id=1286%3A6710 on the very right in the yellow area of "Latest Eng Ready"

File: components/ApplicationForm/ApplicationForm.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                        </tbody>
+                      </table>
+                    </div>
+
+                    <div className="flex flex-col">
+                      <label htmlFor="referral" className="mb-2 font-semibold">
+                        Referral
+                      </label>
+                      <input
+                        placeholder="Optional"
```
Comment (mikefortney) [inline components/ApplicationForm/ApplicationForm.js:208]: This is labeled as optional but is actually required. Please make it optional.

File: components/ApplicationForm/ApplicationForm.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                        onBlur={handleBlur}
+                        value={values.extraInfo}
+                        required
+                      />
+                    </div>
+                    <button
+                      className="p-3 mx-auto border border-indigo-600 rounded-xl text-left font-semibold"
+                      style={{ color: "#5F75EE" }}
+                    >
+                      Add link
```
Comment (mikefortney) [inline components/ApplicationForm/ApplicationForm.js:171]: add link button needs action. should this add another input for another link?

File: components/ApplicationForm/ApplicationForm.js [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                      onBlur={handleBlur}
+                      value={values.projectName}
+                    />
+                  </div>
+
+                  <div className="flex flex-col">
+                    <label htmlFor="pitch-tweet" className="mb-2 font-semibold">
+                      Pitch us your project in a tweet
+                    </label>
+                    <textarea
```
Comment (mikefortney) [inline components/ApplicationForm/ApplicationForm.js:74]: this needs a 280 character limit (also needs to be called out in label)

File: components/ApplicationForm/ApplicationForm.js (hunk 2 above)
Comment (mikefortney) [reply components/ApplicationForm/ApplicationForm.js:171]: nm i just noticed your note about this needing to be added

**Review by mikefortney — CHANGES_REQUESTED** [review]
this is more aesthetic but the first two inputs (and referral) have a nice border/shadow at the bottom but the textareas and link input do not. can they be added?

File: components/ApplicationForm/ApplicationForm.js (hunk 2 above)
Comment (kareemarab) [reply components/ApplicationForm/ApplicationForm.js:171]: yeah, I'll implement that and keep a part of the PR actually. Just realized it's not that big a task.

File: components/ApplicationForm/ApplicationForm.js (hunk 3 above)
Comment (kareemarab) [reply components/ApplicationForm/ApplicationForm.js:74]: yup, fixing that too. Thanks!

**Comment by kareemarab** [comment]
> Interesting. But I agree. The design you are looking at doesn't diverge too far from the implementation, that's fine imo. Are we looking at different figmas maybe? I am in https://www.figma.com/file/CPfZkeob1dtxStjSuUUXDs/HyperscalePaid?node-id=1286%3A6710 on the very right in the yellow area of "Latest Eng Ready"

I was looking @ the "current design" under Dec. 13. I'll double check w/ core-design and get back to you on this.

**Review by mikefortney — APPROVED** [review]

**Comment by tomknig** [comment]
> > Interesting. But I agree. The design you are looking at doesn't diverge too far from the implementation, that's fine imo. Are we looking at different figmas maybe? I am in https://www.figma.com/file/CPfZkeob1dtxStjSuUUXDs/HyperscalePaid?node-id=1286%3A6710 on the very right in the yellow area of "Latest Eng Ready"
> 
> 
> 
> @tomknig I was looking @ the "current design" under Dec. 13. I'll double check w/ core-design and get back to you on this.

Ok. Design has responded. Looks like this PR is not yet implementing the latest design: https://discord.com/channels/916751469678579784/918664975998930954/920918234063061012

How do we want to proceed? Implementing latest design as a follow up or adapt as a part of this PR?

**Comment by kurdybacha** [comment]
Let's merge it and we can address in the follow up PR.
Nice work!

**Pull request merged**